shader/draw_leaf.wgsl

```diff
@@ -74,184 +72,198 @@ fn main(
     // Two barriers can be eliminated if we use separate shared arrays
     // for prefix and intra-workgroup prefix sum.
     workgroupBarrier();
-    var m = sh_scratch[0];
-    workgroupBarrier();
-    let tag_word = read_draw_tag_from_scene(ix);
-    agg = map_draw_tag(tag_word);
-    sh_scratch[local_id.x] = agg;
-    for (var i = 0u; i < firstTrailingBit(WG_SIZE); i += 1u) {
+    var prefix = sh_scratch[0];
+
+    let num_blocks_total = (config.n_drawobj + (WG_SIZE - 1u)) / WG_SIZE;
+    let n_blocks_base = num_blocks_total / WG_SIZE;
+    let remainder = num_blocks_total % WG_SIZE;
+    let first_block = n_blocks_base * wg_id.x + min(wg_id.x, remainder);
+    let n_blocks = n_blocks_base + u32(wg_id.x < remainder);
```

Comment:
If I understand lines 79-81 correctly, they distribute the remainder across workgroups to load balance? I would add a comment here that clarifies the block/workgroup/thread assignment.

I would also consider moving these calculations to a helper somewhere in shared/ so that they can be reused in `shader/draw_reduce.wgsl` (though I don't feel strongly about this).